File: src/DNI.Services/ShowList/Show.cs
using System;
using System.Collections.Generic;

namespace DNI.Services.ShowList {
    public class Show {
        public string AudioUrl { get; set; }

        public string Title { get; set; }

        public string Summary { get; set; }

        public string ShowNotes { get; set; }

        public string ShowNotesHtml { get; set; }

        public DateTime PublishedTime { get; set; }

        public decimal? Version { get; set; }

        public string Slug { get; set; }

        public string ImageUrl { get; set; }

        public string PodcastPageUrl { get; set; }

        public string Duration { get; set; }

        public long? DurationInSeconds { get; set; }

        public IEnumerable<string> Keywords { get; set; }
    }
}

File: src/DNI.Web/src/app/model/show.ts
import { ShowKeyword } from './show-keyword';

export class Show {
    public title: string;
    public summary: string;
    public showNotes: string;
    public showNotesHtml: string;
    public publishedTime: Date;
    public audioUrl: string;
    public version: number; // decimal
    public imageUrl: string;
    public slug: string;
    public podcastPageUrl: string;
    public duration: string;
    public durationInSeconds: number;
    public keywords: string[];

    public getKeywords(): ShowKeyword[] {
        return this.keywords
            .map(s => new ShowKeyword(s, null));
    }
}
